site/en/docs/extensions/reference/offscreen/index.md

```diff
@@ -46,11 +46,50 @@ chrome.offscreen.createDocument({
 chrome.offscreen.closeDocument()
 ```
 
-For a complete extension, see the [offscreen-clipboard demo][gh-offscreen-clipboard] on GitHub.
+The following example shows how to check for existing offscreen documents. The `hasOffscreenDocument` method uses [clients.matchAll()](https://developer.mozilla.org/docs/Web/API/Clients/matchAll) to find existing offscreen documents.
 
-[api-runtime]: /docs/extensions/reference/runtime/
-[api-windows]: /docs/extensions/reference/windows/
-[doc-manifest]: /docs/extensions/mv3/manifest/
-[gh-offscreen-clipboard]: https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/cookbook.offscreen-clipboard-write
-[offscreen-reason]: /docs/extensions/reference/offscreen/#type-Reason
+```js
+async function hasOffscreenDocument(path) {
+  // Check all windows controlled by the service worker if one 
+  // of them is the offscreen document with the given path
```

Comment:
Maybe something like this?

```
// Check to see if there is a context associated with the service worker matching the expected path.
```

I'm reluctant to say `check all windows` because I think clients can also be workers etc.

Reply:
You can filter by type of client and the default is `window` https://developer.mozilla.org/en-US/docs/Web/API/Clients/matchAll

Reply:
Oh, TIL!